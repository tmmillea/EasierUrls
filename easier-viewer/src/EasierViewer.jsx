import React, {useEffect} from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Button from '@material-ui/core/Button';
import Link from '@material-ui/core/Link';
import Paper from '@material-ui/core/Paper';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import TableCell from '@material-ui/core/TableCell';
import { TableContainer } from '@material-ui/core';
import TextField from '@material-ui/core/TextField';

const rootUrl = 'http://127.0.0.1:8080/easier/';
const useStyles = makeStyles(theme => ({
  root: {
    '& > *': {
      margin: theme.spacing(1),
    },
  },
  field: {
    margin: theme.spacing(1),
    width: 200,
  },
  paper: {
    width: '100%',
    marginBottom: theme.spacing(2),
  },
}));

function UrlInput(props) {
  const {classes, url, setUrl, error, submit} = props;
  const handleChange = (event) => {
    setUrl(event.target.value);
  }
  return (
    <div className={classes.root}>
      <TextField error={error} className={classes.field} value={url} id="urlInput" label="Url Input" onChange={handleChange}/>
      <Button variant="contained" color="primary" onClick={submit}>Create Url</Button>
    </div>
  );
}

function EasierViewer() {
  const classes = useStyles();
  const [url, setUrl] = React.useState("");
  const [links, setLinks] = React.useState([]);
  const [error, setError] = React.useState(false);

  const loadLinks = () => {
    fetch(rootUrl).then((response) => response.json().then((jsonData) => {
      const smallUrlList = jsonData.smallUrlList;
      const newLinks = [];
      smallUrlList.map((urlData) => {
        newLinks.push([rootUrl + urlData.urlPath, urlData.data.target]);
      });
      setLinks(newLinks);
    }).catch(() => {
      setLinks([]);
      setError(true);
    }));
  }

  useEffect(() => {
    loadLinks();
  }, [links]);

    const submit = () => {
      const init = {
        method: 'PUT',
        body: url,
        headers: {
          'Access-Control-Allow-Origin': '*',
        }
      };
      fetch(rootUrl, init)
        .then((response) => {
          if(response.ok) {
            loadLinks();
            setError(false);
            setUrl("");
          } else {
            setError(true);
          }
        })
    }    

  return (
    <div className={classes.root}>
      <Paper className={classes.paper}>
        <UrlInput 
          classes={classes}
          url={url}
          setUrl={setUrl}
          submit={submit}
          error={error}
        />  
        <TableContainer>
          <Table className={classes.table}>
            <TableHead>
              <TableRow>
                <TableCell>Easier Url Link</TableCell>
                <TableCell>Destination</TableCell>
              </TableRow>
            </TableHead>
            <TableBody>
              {links.map((link) => {
                return(
                  <TableRow>
                    <TableCell>
                      <Link href={link[1]}>{link[0]}</Link>
                    </TableCell>
                    <TableCell>{link[1]}</TableCell>
                  </TableRow>
                );
              })}
            </TableBody>
          </Table>  
        </TableContainer>
      </Paper>
    </div>
  );
}

export default EasierViewer;
