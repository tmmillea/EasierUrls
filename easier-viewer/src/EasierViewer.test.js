import React from 'react';
import TestRenderer from 'react-test-renderer';
import EasierViewer from './EasierViewer';
const {act} = TestRenderer;
test('empty snapshot', () => {
  const render = TestRenderer.create(<EasierViewer/>);
  expect(render.toJSON()).toMatchSnapshot();
  render.unmount();
});
test('Initialized', async () => {
  const urlList = {
    smallUrlList: [
    {
      urlPath: "VhQLtXr9h4YW",
      data: {
        target: "https://google.com",
        remaining: 10,
      }
    },
    {
      urlPath: "1sODnR6Vtcbd",
      data: {
        target: "https://intuit.com",
        remaining: 10,
      }
    },
    {
      urlPath: "fQTC51Aagwfj",
      data: {
        target: "https://amazon.com",
        remaining: 10,
      }
    },]
  };
  fetchMock.mockResponse(JSON.stringify(urlList));
  let wrapper;
  await act(() => {
    wrapper = TestRenderer.create(<EasierViewer/>);
  })
  expect(wrapper.toJSON()).toMatchSnapshot();
  wrapper.unmount();
});
