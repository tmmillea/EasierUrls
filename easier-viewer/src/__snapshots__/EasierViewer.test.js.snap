// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Initialized 1`] = `
<div
  className="makeStyles-root-143"
>
  <div
    className="MuiPaper-root makeStyles-paper-145 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      className="makeStyles-root-143"
    >
      <div
        className="MuiFormControl-root MuiTextField-root makeStyles-field-144"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
          data-shrink={false}
          htmlFor="urlInput"
          id="urlInput-label"
        >
          Url Input
        </label>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiInput-input"
            disabled={false}
            id="urlInput"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="text"
            value=""
          />
        </div>
      </div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Create Url
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      className="MuiTableContainer-root"
    >
      <table
        className="MuiTable-root"
      >
        <thead
          className="MuiTableHead-root"
        >
          <tr
            className="MuiTableRow-root MuiTableRow-head"
          >
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              Easier Url Link
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              Destination
            </th>
          </tr>
        </thead>
        <tbody
          className="MuiTableBody-root"
        />
      </table>
    </div>
  </div>
</div>
`;

exports[`empty snapshot 1`] = `
<div
  className="makeStyles-root-1"
>
  <div
    className="MuiPaper-root makeStyles-paper-3 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      className="makeStyles-root-1"
    >
      <div
        className="MuiFormControl-root MuiTextField-root makeStyles-field-2"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
          data-shrink={false}
          htmlFor="urlInput"
          id="urlInput-label"
        >
          Url Input
        </label>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiInput-input"
            disabled={false}
            id="urlInput"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="text"
            value=""
          />
        </div>
      </div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Create Url
        </span>
      </button>
    </div>
    <div
      className="MuiTableContainer-root"
    >
      <table
        className="MuiTable-root"
      >
        <thead
          className="MuiTableHead-root"
        >
          <tr
            className="MuiTableRow-root MuiTableRow-head"
          >
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              Easier Url Link
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              Destination
            </th>
          </tr>
        </thead>
        <tbody
          className="MuiTableBody-root"
        />
      </table>
    </div>
  </div>
</div>
`;
